CMakeLists.txt

```diff
@@ -746,20 +746,15 @@ if (WIN32)
     list(APPEND LINK_LIBRARIES version.lib shlwapi.lib odbc32.lib odbccp32.lib)
 else()
     if (ISPC_STATIC_LINK)
-        list(APPEND LINK_LIBRARIES tinfo.a curses.a)
+      find_library(CURSES_LIB REQUIRED NAMES
+                     "${CMAKE_STATIC_LIBRARY_PREFIX}curses${CMAKE_STATIC_LIBRARY_SUFFIX}"
+                     "${CMAKE_STATIC_LIBRARY_PREFIX}ncurses${CMAKE_STATIC_LIBRARY_SUFFIX}")
+        find_library(TINFO_LIB REQUIRED NAMES "${CMAKE_STATIC_LIBRARY_PREFIX}tinfo${CMAKE_STATIC_LIBRARY_SUFFIX}")
+        list(APPEND LINK_LIBRARIES ${TINFO_LIB})
     else()
-        find_package(Curses REQUIRED)
-        if (CURSES_EXTRA_LIBRARY)
-            # this contains the tinfo library, if found
-            list(APPEND LINK_LIBRARIES ${CURSES_EXTRA_LIBRARY})
-        endif()
-        # Recommended way to get Curses library is to use CURSES_LIBRARIES variable
-        # https://cmake.org/cmake/help/v3.27/module/FindCurses.html
-        # but it contains both libform.so and libcurses.so.
-        # We need libcurses.so only, so exclude libform.so from the list.
-        list(FILTER CURSES_LIBRARIES EXCLUDE REGEX ".*libform.*")
-        list(APPEND LINK_LIBRARIES ${CURSES_LIBRARIES})
+        find_library(CURSES_LIB REQUIRED NAMES curses)
```

Comment:
This change is not equivalent to what was before. What about tinfo library in case of dynamic linking? 
Even if it works in our environment I recommend double check it. There was an issue related to that.

Reply:
Right. I have changed this to follow the LLVM logic [here](https://github.com/llvm/llvm-project/blob/main/llvm/cmake/modules/FindTerminfo.cmake)

Reply:
Have the set of link libraries changed for release package or stayed the same as it was?

Reply:
The release package used and uses the path under `ISPC_STATIC_LINK`. So, the `find_package(Curses REQUIRED)` and the following logic has never used in the release build. The set of linked libraries hasn't changed.